### PR TITLE
Describe API groups and spec/status for simple kinds

### PR DESCRIPTION
**Pull request by smarterclayton**
Clarifies a few points folks have asked.

### CODE REVIEW

**Comment by smarterclayton** [comment]
@bgrant0607 @erictune clarifying both use of spec / status on SubjectAccessReview as well as group naming.

**Comment by MHBauer** [comment]
Do you want to bump the date on the top?

File: contributors/devel/api-conventions.md [hunk 1]
```diff
@@ -156,6 +168,11 @@ API endpoints may deviate from the strict REST pattern and return resources that
 are not singular JSON objects, such as streams of JSON objects or unstructured
 text log data.
 
+Many simple kinds are used across all API groups and are thus considered part
```
Comment (MHBauer) [inline contributors/devel/api-conventions.md]: Such as what types? Where would I go to find them? Why would I care? Is it a correctness issue, or a code re-use best practice issue?

File: contributors/devel/api-conventions.md [hunk 2]
```diff
@@ -276,6 +293,13 @@ cannot vary from the user's desired intent MAY have only "spec", and MAY rename
 Objects that contain both spec and status should not contain additional
 top-level fields other than the standard metadata fields.
 
+Some objects which are not persisted in the system may choose to add spec and
```
Comment (MHBauer) [inline contributors/devel/api-conventions.md]: Is there an example of this?

**Comment by smarterclayton** [comment]
Updated based on the comments

**Comment by bgrant0607** [comment]
Thanks. Merge and iterate.

**Pull request merged**